api/api_gateway/routers/scans.py

```diff
@@ -133,3 +134,28 @@ def retrieve_assemblyline_messages(session):
             break
 
     return scan_results
+
+
+def determine_verdict(provider, value):
+    if provider is None or value is None:
+        log.error(
+            f"Provider({provider}) and Value({value}) are required to calculate scan verdicts"
+        )
+        return ScanVerdicts.ERROR.value
+
+    if provider == ScanProviders.ASSEMBLYLINE.value:
+        if not isinstance(value, int):
+            return ScanVerdicts.ERROR.value
+        if (
+            value == -1000 or 0 <= value <= 299
+        ):  # Ignore informational ratings and merge with clean
+            return ScanVerdicts.CLEAN.value
+        elif 300 <= value <= 999:
+            return ScanVerdicts.SUSPICIOUS.value
+        elif value >= 1000:
+            return ScanVerdicts.MALICIOUS.value
+        else:
+            return ScanVerdicts.ERROR.value
```

Comment:
Would it make sense to set this to some other verdict (e.g. `ScanVerdicts.UNKNOWN`) just so we'd be able to separate them out from error cases?   I'm thinking of a scenario where AssemblyLine is updated in the future and there's suddenly a new error code range.

Or is it safe to say that the ranges not covered by the `if/else` above are legit errors?

Reply:
i like it, it'll help with future work related to automatically retrying when errors are detected. Having these flagged as unknown will help us find the issue before attempting to retry.